internal/bminventory/inventory.go

```diff
@@ -4064,9 +4064,11 @@ func (b *bareMetalInventory) checkFileForDownload(ctx context.Context, clusterID
 	switch fileName {
 	case constants.Kubeconfig:
 		err = clusterPkg.CanDownloadKubeconfig(cluster)
+	case constants.KubeadminPassword:
+		fallthrough
 	case constants.KubeconfigNoIngress:
-		agentInstaller := b.Config.ClusterTLSCertOverrideDir != ""
-		err = clusterPkg.CanDownloadKubeconfigNoIngress(cluster, agentInstaller)
+		agentInstaller := b.installerInvoker == "agent-installer"
+		err = clusterPkg.CanDownloadKubeconfigFiles(cluster, fileName, agentInstaller)
```

Comment:
```suggestion
		err = clusterPkg.CanDownloadKubeconfigFiles(cluster, fileName, b.installerInvoker)
```

nit: embedding the logic in `CanDownloadKubeconfigFiles` could save one line for every potential user of the method

Reply:
Changed.